### PR TITLE
feat(service.go): Panic catcher not working

### PR DESCRIPTION
**Pull request by learnjin**
If my service handler has a panic, the server should give a 500, instead of crashing.

### CODE REVIEW

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: dab9f03ca1c52c21a6f4e8baf6b0e8bfafe4faa0
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/microdevs/missy

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: dab9f03ca1c52c21a6f4e8baf6b0e8bfafe4faa0
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: dde3f0919b07c814e9379cfee8a5ac8286157468
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: d382ec493b4e314812148ac54f33509580c00d79
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/microdevs/missy

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: dab9f03ca1c52c21a6f4e8baf6b0e8bfafe4faa0
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: dde3f0919b07c814e9379cfee8a5ac8286157468
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: d382ec493b4e314812148ac54f33509580c00d79
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/microdevs/missy

---
This message was auto-generated by https://gitcop.com

**Comment by slot** [comment]
@learnjin check the implementation, should work now

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: f86c3d623562d393a7c7e704404a180ae16b08bd
  * Invalid type. Valid types are **feat, fix, docs, style, refactor, perf, test, chore, revert**

Guidelines are available at https://github.com/microdevs/missy

---
This message was auto-generated by https://gitcop.com

File: service/service.go [hunk 1]
```diff
@@ -137,14 +144,22 @@ func (s *Service) prepareBeforeStart() {
...
 func (s *Service) HandleFunc(pattern string, handler func(http.ResponseWriter, *http.Request)) *mux.Route {
 	return s.Handle(pattern, http.HandlerFunc(handler))
 }
 
-// Handle
+// Handle is a wrapper around the original Go handle func with logging recovery and metrics
 func (s *Service) Handle(pattern string, originalHandler http.Handler) *mux.Route {
 	h := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		defer func() {
+			if err := recover(); err != nil {
```
Comment (learnjin) [inline service/service.go]: OK. So the `recover` is the trick. I assume this is always called, even if there's no pani, but then the `err` is nil and the rest won't get executed. Is that about right?

Comment (slot) [reply service/service.go]: Yes. see https://blog.golang.org/defer-panic-and-recover

File: service/service.go [hunk 2]
```diff
@@ -127,6 +129,10 @@ func (s *Service) Start() {
 	log.Infof("Service stopped gracefully.")
 }
 
+func (s *Service) Shutdown() {
```
Comment (levrik) [inline service/service.go]: Why was this added?

File: service/service.go [hunk 3]
```diff
@@ -127,6 +129,11 @@ func (s *Service) Start() {
 	log.Infof("Service stopped gracefully.")
 }
 
+// Shutdown allows to stop the HTTP Server gracefully
+func (s *Service) Shutdown() {
```
Comment (levrik) [inline service/service.go]: Why was this added?
